File: src/simulation/space/resource/dependency/ResourceDependency.java
package simulation.space.resource.dependency;

import simulation.space.tile.Tile;
import simulation.space.resource.Resource;

public interface ResourceDependency {
    double satisfactionPercent(Tile tile, Resource resource);//TODO sometimes null, fix this
    boolean isNecessary();
    boolean isPositive();
    boolean isResourceNeeded();
    boolean hasNeeded(Tile tile);
}

File: src/simulation/space/resource/dependency/ResourceDependency.kt
package simulation.space.resource.dependency

import simulation.space.resource.Resource
import simulation.space.tile.Tile

interface ResourceDependency {
    fun satisfactionPercent(tile: Tile, resource: Resource): Double
    val isNecessary: Boolean
    val isPositive: Boolean
    val isResourceNeeded: Boolean
    fun hasNeeded(tile: Tile): Boolean
}
